rasa/nlu/training_data/training_data.py

```diff
@@ -138,7 +138,18 @@ def persist(
             os.makedirs(dir_name)
 
         data_file = os.path.join(dir_name, filename)
-        write_to_file(data_file, self.as_json(indent=2))
+
+        fformat = "md" if filename.endswith(".md") else "json"
```

Comment:
I think this function might be called when persisting models to make the training data part of the tar.gz

When persisting in the model, we should always use the same format (`json`).